boards/DIM/Inc/App/App_BoardStatusLeds.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include "App_BoardEnum.h"
+#include "App_SharedExitCode.h"
+
+/**
+ * Allocate and initialize a group of board status LEDs
+ * @param turn_on_dim_red A function that can be called to turn the DIM status
+ *                        LED red
+ * @param turn_on_dim_green A function that can be called to turn the DIM status
+ *                          LED green
+ * @param turn_on_dim_blue A function that can be called to turn the DIM status
+ *                         LED blue
+ * @param turn_on_dim_red A function that can be called to turn the DCM status
+ *                        LED red
+ * @param turn_on_dim_green A function that can be called to turn the DCM status
+ *                          LED green
+ * @param turn_on_dim_blue A function that can be called to turn the DCM status
+ *                         LED blue
+ * @param turn_on_dim_red A function that can be called to turn the FSM status
+ *                        LED red
+ * @param turn_on_dim_green A function that can be called to turn the FSM status
+ *                          LED green
+ * @param turn_on_dim_blue A function that can be called to turn the FSM status
+ *                         LED blue
+ * @param turn_on_dim_red A function that can be called to turn the BMS status
+ *                        LED red
+ * @param turn_on_dim_green A function that can be called to turn the BMS status
+ *                          LED green
+ * @param turn_on_dim_blue A function that can be called to turn the BMS status
+ *                         LED blue
+ * @param turn_on_dim_red A function that can be called to turn the PDM status
+ *                        LED red
+ * @param turn_on_dim_green A function that can be called to turn the PDM status
+ *                          LED green
+ * @param turn_on_dim_blue A function that can be called to turn the PDM status
+ *                         LED blue
+ * @return The created group of board status LEDs, whose ownership is given to
+ *         the caller
+ */
+struct BoardStatusLeds *App_BoardStatusLeds_Create(
+    void (*turn_on_dim_red)(void),
+    void (*turn_on_dim_green)(void),
+    void (*turn_on_dim_blue)(void),
+    void (*turn_on_dcm_red)(void),
+    void (*turn_on_dcm_green)(void),
+    void (*turn_on_dcm_blue)(void),
+    void (*turn_on_fsm_red)(void),
+    void (*turn_on_fsm_green)(void),
+    void (*turn_on_fsm_blue)(void),
+    void (*turn_on_bms_red)(void),
+    void (*turn_on_bms_green)(void),
+    void (*turn_on_bms_blue)(void),
+    void (*turn_on_pdm_red)(void),
+    void (*turn_on_pdm_green)(void),
+    void (*turn_on_pdm_blue)(void));
```

Comment:
At another extreme, we could have:
```
struct SharedLed *App_SharedLed_Create(
    void (*enable)(void),
    void (*disable)(void));

struct BoardStatusLedGroup* App_BoardStatusLedGroup_Create(
     struct SharedLed* green, struct SharedLed* red, struct SharedLed* blue);

struct BoardStatusDisplay* App_BoardStatusDisplay_Create(
     struct BoardStatusLedGroup fsm_leds,
     ....
     );
```

And there are definitely halfway options between this suggestion and what's currently in code. 

I think it's OK as-is, but just wanted to present these options to see what you think. Personally I'd lean towards what I have above, but again, happy to approve it as-is.

Reply:
That's fine with me, let me spend some time refactoring it

Reply:
I think I came up with a even cleaner solution. Take a look at the new changes and let me know what you think!